File: sns/src/test/scala/akka/stream/alpakka/sns/scaladsl/Examples.scala
/*
 * Copyright (C) 2016-2018 Lightbend Inc. <http://www.lightbend.com>
 */

package akka.stream.alpakka.sns.scaladsl

import akka.Done
import akka.actor.ActorSystem
import akka.stream.ActorMaterializer
import akka.stream.scaladsl.{Sink, Source}
import com.amazonaws.auth.{AWSStaticCredentialsProvider, BasicAWSCredentials}
import com.amazonaws.services.sns.model.PublishRequest
import com.amazonaws.services.sns.{AmazonSNSAsync, AmazonSNSAsyncClientBuilder}

import scala.concurrent.Future

object Examples {

  //#init-client
  val credentials = new BasicAWSCredentials("x", "x")
  implicit val snsClient: AmazonSNSAsync =
    AmazonSNSAsyncClientBuilder.standard().withCredentials(new AWSStaticCredentialsProvider(credentials)).build()
  //#init-client

  //#init-system
  implicit val system: ActorSystem = ActorSystem()
  implicit val mat: ActorMaterializer = ActorMaterializer()
  //#init-system

  //#use-sink
  val stringSink: Future[Done] =
    Source
      .single("message")
      .runWith(SnsPublisher.sink("topic-arn"))

  val publishRequestSink: Future[Done] =
    Source
      .single(new PublishRequest().withMessage("message"))
      .runWith(SnsPublisher.publishSink("topic-arn"))
  //#use-sink

  //#use-flow
  val stringFLow: Future[Done] =
    Source
      .single("message")
      .via(SnsPublisher.flow("topic-arn"))
      .runWith(Sink.ignore)

  val publishRequestFlow: Future[Done] =
    Source
      .single(new PublishRequest().withMessage("message"))
      .via(SnsPublisher.publishFlow("topic-arn"))
      .runWith(Sink.ignore)
  //#use-flow

}

File: sns/src/test/java/akka/stream/alpakka/sns/javadsl/Examples.java
/*
 * Copyright (C) 2016-2018 Lightbend Inc. <http://www.lightbend.com>
 */

package akka.stream.alpakka.sns.javadsl;

import akka.Done;
import akka.actor.ActorSystem;
import akka.stream.ActorMaterializer;
import akka.stream.javadsl.Sink;
import akka.stream.javadsl.Source;
import com.amazonaws.auth.AWSStaticCredentialsProvider;
import com.amazonaws.auth.BasicAWSCredentials;
import com.amazonaws.services.sns.AmazonSNSAsync;
import com.amazonaws.services.sns.AmazonSNSAsyncClientBuilder;
import com.amazonaws.services.sns.model.PublishRequest;

import java.util.concurrent.CompletionStage;

public class Examples {

  // #init-client
  BasicAWSCredentials credentials = new BasicAWSCredentials("x", "x");
  AmazonSNSAsync snsClient =
      AmazonSNSAsyncClientBuilder.standard()
          .withCredentials(new AWSStaticCredentialsProvider(credentials))
          .build();
  // #init-client

  // #init-system
  ActorSystem system = ActorSystem.create();
  ActorMaterializer materializer = ActorMaterializer.create(system);
  // #init-system

  // #use-sink
  CompletionStage<Done> stringSink =
      Source.single("message")
          .runWith(SnsPublisher.createSink("topic-arn", snsClient), materializer);

  CompletionStage<Done> publishRequestSink =
      Source.single(new PublishRequest().withMessage("message"))
          .runWith(SnsPublisher.createPublishSink("topic-arn", snsClient), materializer);
  // #use-sink

  // #use-flow
  CompletionStage<Done> stringFlow =
      Source.single("message")
          .via(SnsPublisher.createFlow("topic-arn", snsClient))
          .runWith(Sink.ignore(), materializer);

  CompletionStage<Done> publishRequestFlow =
      Source.single(new PublishRequest().withMessage("message"))
          .via(SnsPublisher.createPublishFlow("topic-arn", snsClient))
          .runWith(Sink.ignore(), materializer);
  // #use-flow

}
